File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
vector<int> twoSum(vector<int> &nums, int target) {
    vector<int> res(2);
    int cnt = 0;
    for (auto i: nums) {
        cnt++;
    }
    for (int i = 0; i < cnt; ++i) {
        int x = target - nums[i];
        bool bol = binary_search(nums.begin(), nums.end(), x);
        if (bol) {
            int j = lower_bound(nums.begin(), nums.end(), x) - nums.begin();
            if (i==j){
                res[0] = i+1;
                res[1] = i+2;
            }else{
                res[0] = i+1;
                res[1] = j+1;
            }
            return res;
        }
    }
    return res;
}
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length-1;
        
        while(l<r){
            int s = (numbers[l] + numbers[r]);
            if(s == target){
                return new int [] {l+1,r+1};
            }else if(s<target){
                l++;
            }else{
                r--;
            }
        }
         return new int [] {};
    }
}